pallets/xcm-core-buyer/src/lib.rs

```
@@ -0,0 +1,456 @@
+            if assigned_collators.is_empty() {
+                return Err(Error::<T>::NoAssignedCollators.into());
+            }
+
+            Self::on_collator_instantaneous_core_requested(para_id)
+        }
+
+        #[pallet::call_index(2)]
+        #[pallet::weight(T::WeightInfo::set_xcm_weights())]
+        pub fn set_xcm_weights(
+            origin: OriginFor<T>,
+            xcm_weights: Option<XcmWeightsTy<T>>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            if let Some(xcm_weights) = xcm_weights {
+                XcmWeights::<T>::put(xcm_weights);
+            } else {
+                XcmWeights::<T>::kill();
+            }
+
+            Ok(())
+        }
+    }
+
+    impl<T: Config> Pallet<T> {
+        /// Returns the interior multilocation for this container chain para id. This is a relative
+        /// multilocation that can be used in the `descend_origin` XCM opcode.
+        pub fn interior_multilocation(para_id: ParaId) -> InteriorMultiLocation {
+            let container_chain_account = T::GetParathreadAccountId::convert(para_id);
+            let account_junction = Junction::AccountId32 {
+                id: container_chain_account,
+                network: None,
+            };
+
+            InteriorMultiLocation::X1(account_junction)
+        }
+
+        /// Returns a multilocation that can be used in the `deposit_asset` XCM opcode.
+        /// The `interior_multilocation` can be obtained using `Self::interior_multilocation`.
+        pub fn absolute_multilocation(
+            interior_multilocation: InteriorMultiLocation,
+        ) -> MultiLocation {
+            let relay_chain = MultiLocation::parent();
+            let context = Parachain(T::SelfParaId::get().into()).into();
+            let mut reanchored: MultiLocation = interior_multilocation.into();
+            reanchored
+                .reanchor(&relay_chain, context)
+                .expect("reanchor failed");
```

Remove this expect, let's just return a pallet error